### DISCUSSION TITLE
Some details in GROOPS Parallel processing

### DISCUSSION THREAD

**Discussion by FFFFFG-IGG** [groops-devs/groops#139] [Q&A] [answered]
Hello developers. According to Stresser's article, large net analysis can run in parallel through ppp, when satellite parameters are fixed. However, when I run GNSS net analysis as TUG provides in https://ftp.tugraz.at/outgoing/ITSG/groops/scenario/scenarioGnssNetwork.zip with groopsMPI, stations of the net are still run one by one, and I do not find any clue how to delploy them to run parallel. Could you give me some suggestions about that?

**Comment by tmayerguerr** [comment, marked as answer]
Hi, 

There is a small section in our cookbook [GNSS satellite orbit determination and station network analysis](https://groops-devs.github.io/groops/html/cookbook.gnssNetwork.html#advanced) about this topic.
After processing a core network, the key element is [gnssProcessingStep->forEachReceiverSeparately](https://groops-devs.github.io/groops/html/gnssProcessingStepType.html#forEachReceiverSeparately).

Please see also the answer in [discussions/128](https://github.com/groops-devs/groops/discussions/128#discussioncomment-4256140) from Patrick.

regards
Torsten

**Reply by FFFFFG-IGG** [reply]
> Hi,
> 
> There is a small section in our cookbook [GNSS satellite orbit determination and station network analysis](https://groops-devs.github.io/groops/html/cookbook.gnssNetwork.html#advanced) about this topic. After processing a core network, the key element is [gnssProcessingStep->forEachReceiverSeparately](https://groops-devs.github.io/groops/html/gnssProcessingStepType.html#forEachReceiverSeparately).
> 
> Please see also the answer in [discussions/128](https://github.com/groops-devs/groops/discussions/128#discussioncomment-4256140) from Patrick.
> 
> regards Torsten

Thanks, Torsten. I did use gnssProcessingStep->forEachReceiverSeparatelty but the log file shows that station still run one by one. Did I miss some settings in .xml somewhere?Acorrding https://github.com/groops-devs/groops/discussions/128#discussioncomment-4256140 from Patrick, mpi may conflict with OpenBlas, I set the same variables in .bashrc(as shown in pictures bellow):
![image](https://user-images.githubusercontent.com/59271400/205539795-6046b0e4-d9d0-4cf9-9918-8793c540efda.png)
And the log file still show that stations are run one-by-one:
![image](https://user-images.githubusercontent.com/59271400/205540078-a5d9a9b6-f931-48a6-9eaa-66c5127b7e52.png)
![image](https://user-images.githubusercontent.com/59271400/205540280-643addb4-2052-45c4-b860-aadb899a9d9f.png)
It's obvious that station 'aruc' run after 'abmf', which confirms that stations are not processed in parallel.

Some notes:
1.Input in command line: mpiexec -n 4 groopsMpi -l log.txt 03Network.xml
2.Satellite params are fixed(sp3,clock,obx and osb are gotten from wum), I did not set core net, just to test parallel-ppp->whole net procedure.
3.log.txt and 03Network.xml are in the attachment. 

Could you please offer some help?
Many thanks!
Yongzhao Fan.
[attachment.zip](https://github.com/groops-devs/groops/files/10150356/attachment.zip)

**Comment by tmayerguerr** [comment]
Dear Yongzhao Fan

Everthing works as expected. You start GROOPS with 4 cores to process 12 stations. Therefore every core process 3 stations (4 stations in parallel). To avoid spamming the log only the reults of the first core is displayed.

```
2022-12-05 10:55:08 Status  === Starting GROOPS with 4 processes ===
2022-12-05 10:55:59 Status  === for each receiver separately ============================
2022-12-05 10:55:59 Info      12 receivers selected
2022-12-05 10:55:59 Info      Only results of a subset of stations are displayed in the following
2022-12-05 10:55:59 Status  === select single receiver (abmf) ===========================
2022-12-05 10:59:06 Status  === select single receiver (aruc) ===========================
2022-12-05 11:02:30 Status  === select single receiver (bogt) ===========================
```
regards
Torsten

**Reply by FFFFFG-IGG** [reply]
Thanks for your kind help!
